server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java

```diff
@@ -62,81 +60,109 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class FileSystemInitializer {
+public class FileSystemInitializer {
   private static final String TABLE_TABLETS_TABLET_DIR = "table_info";
   private static final Logger log = LoggerFactory.getLogger(FileSystemInitializer.class);
+  private static final Text SPLIT_POINT =
+      MetadataSchema.TabletsSection.getRange().getEndKey().getRow();
 
   // config only for root table
   private final InitialConfiguration initConfig;
 
-  FileSystemInitializer(InitialConfiguration initConfig, ZooReaderWriter zoo, InstanceId uuid) {
+  public FileSystemInitializer(InitialConfiguration initConfig) {
     this.initConfig = initConfig;
   }
 
-  private static class Tablet {
+  public static class InitialTablet {
     TableId tableId;
     String dirName;
-    Text prevEndRow, endRow;
+    Text prevEndRow, endRow, extent;
     String[] files;
 
-    Tablet(TableId tableId, String dirName, Text prevEndRow, Text endRow, String... files) {
+    InitialTablet(TableId tableId, String dirName, Text prevEndRow, Text endRow, String... files) {
       this.tableId = tableId;
       this.dirName = dirName;
       this.prevEndRow = prevEndRow;
       this.endRow = endRow;
       this.files = files;
+      this.extent = new Text(MetadataSchema.TabletsSection.encodeRow(this.tableId, this.endRow));
     }
+
+    private TreeMap<Key,Value> createEntries() {
+      TreeMap<Key,Value> sorted = new TreeMap<>();
+      Value EMPTY_SIZE = new DataFileValue(0, 0).encodeAsValue();
+      sorted.put(new Key(this.extent, DIRECTORY_COLUMN.getColumnFamily(),
+          DIRECTORY_COLUMN.getColumnQualifier(), 0), new Value(this.dirName));
+      sorted.put(
+          new Key(this.extent, TIME_COLUMN.getColumnFamily(), TIME_COLUMN.getColumnQualifier(), 0),
+          new Value(new MetadataTime(0, TimeType.LOGICAL).encode()));
+      sorted.put(
+          new Key(this.extent, PREV_ROW_COLUMN.getColumnFamily(),
+              PREV_ROW_COLUMN.getColumnQualifier(), 0),
+          MetadataSchema.TabletsSection.TabletColumnFamily.encodePrevEndRow(this.prevEndRow));
+      for (String file : this.files) {
+        var col =
+            new ColumnFQ(MetadataSchema.TabletsSection.DataFileColumnFamily.NAME, new Text(file));
+        sorted.put(new Key(extent, col.getColumnFamily(), col.getColumnQualifier()), EMPTY_SIZE);
+      }
+      return sorted;
+    }
+
+    public Mutation createMutation() {
+      Mutation mutation = new Mutation(this.extent);
+      for (Map.Entry<Key,Value> entry : createEntries().entrySet()) {
+        mutation.put(entry.getKey().getColumnFamily(), entry.getKey().getColumnQualifier(),
+            entry.getKey().getTimestamp(), entry.getValue());
```

Comment:
Should not set the timestamp here.  When its not set the server side will set it using the latest timestamp for the tablet.  Each table has a timestamp for the next insert that it increments on each insert.

```suggestion
        mutation.put(entry.getKey().getColumnFamily(), entry.getKey().getColumnQualifier(),
             entry.getValue());
```